pkg/upgrade/control_plane.go

```diff
@@ -518,3 +524,41 @@ func (u *ControlPlaneUpgrader) etcdctlForPod(ctx context.Context, pod *v1.Pod, a
 
 	return stdout, stderr, err
 }
+
+func (u *ControlPlaneUpgrader) updateAndUploadKubeadmKubernetesVersion() error {
```

Comment:
Do you have any interest in pulling this kind of work out into its own struct, maybe a struct responsible for modifying the kubeadm config? I was thinking if we doin't want to run `kubeadm reset` on the actual nodes themselves we will have to modify the kubeadm config anyway. Might be good to have that struct and tests in place when we get around to fixing that bug as well.

but i could see it as being useful

I don't mind this though, I think this is clear enough

Reply:
I'd prefer to merge this as-is and open issue(s) as needed for future work, if that's ok with you?

Reply:
definitely. Merging.